### ISSUE TITLE
Possibility for a version with an open source optimizer

### ISSUE THREAD

**Issue by vijaiaeroastro** [nzfeng/SWN#3]
Thanks for open-sourcing this amazing work.

Is there a possibility of getting a version of this algorithm that uses an open-source optimizer?

**Comment by nzfeng** [comment]
Hi Vijai,

One solution I considered is using [CoMISo](https://www.graphics.rwth-aachen.de/software/comiso/), which is basically a wrapper around many different solvers (Gurobi, lpsolve, COIN-OR, etc.), and which solver is used depends on what is available on the user's local machine. However, CoMISo was difficult to get working on my own computers (Mac and Linux), requiring a bunch of machine-specific tweaking, so I was hesitant to make it part of the official SWN release.

But I like the idea of replacing the Gurobi-specific code in this project with generic code that just calls whatever solver the user prefers, open-source or otherwise. I liked Gurobi's performance, so it would be nice to keep the options open.  

If you have suggestions for what C++ software to use, I'd be glad to make the changes!

Nicole

**Comment by vijaiaeroastro** [comment]
@nzfeng I do not know of a C++ alternative similar to CoMISo that provides a generic interface to other optimizers. Gurobi is tricky even for academics, sometimes offering licenses that only allow modeling very small problems. If you can provide a fallback with any open-source optimizer, even if there is some performance degradation, it would be useful.

**Comment by nzfeng** [comment]
Yeah, that's a good point. 

I've just pushed a new branch called `open-source-lp` that uses [OR-Tools](https://developers.google.com/optimization/), which I found reasonably straightforward to get installed. OR-Tools provides an interface to several solvers, both open-source and commercial, including the open-source CLP which has [good documented performance](https://blog.zhangzhk.com/files/Comparison-of-Open-Source-Linear-Programming-Solvers.pdf). (OR-Tools also provides an interface to commercial solvers like CPLEX and Gurobi, if someone did still want to use those at some point.)

I've added instructions in the `README` on how to install OR-Tools and how to specify the back-end LP solver.

Let me know if you still have trouble!
